packages/hash/subgraph/src/shared/type-system-patch.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  BaseUri,
+  VersionedUri,
+  validateBaseUri as validateBaseUriGraphApi,
+  validateVersionedUri as validateVersionedUriGraphApi,
+} from "@blockprotocol/type-system-node";
+
+/**
+ * @todo - we are patching these so this package can be used in node and web environments. When we unify the packages
+ *   this will no longer be necessary return types / functionality may differ a bit in the meantime, these are just
+ *   hotfixes.
+ *   https://app.asana.com/0/1201095311341924/1202923896339225/f
+ *
+ * */
+
+export const extractBaseUri = (id: VersionedUri): BaseUri => {
+  return id.match(/(.+\/)v\/(\d+)(.*)/g)![1]!;
+};
+
+export const extractVersion = (id: VersionedUri): number => {
+  return Number(id.match(/(.+\/)v\/(\d+)(.*)/g)![2]!);
```

Comment:
I think the global flag is messing up the regex, in my browser running `"http://test.com/v/31".match(/(.+\/)v\/(\d+)(.*)/g)[1] ` does not produce the correct result, but running `"http://test.com/v/31".match(/(.+\/)v\/(\d+)(.*)/)[1]` does

![image](https://user-images.githubusercontent.com/6988668/202762971-61c2016b-52bb-4b18-b8e5-599a0c505bad.png)